core/src/main/scala/org/typelevel/sbt/TypelevelPlugin.scala

```diff
@@ -0,0 +1,30 @@
+package org.typelevel.sbt
+
+import sbt._
+import sbtghactions.GitHubActionsPlugin
+
+object TypelevelPlugin extends AutoPlugin {
+
+  override def requires =
+    TypelevelKernelPlugin && TypelevelSettingsPlugin && TypelevelVersioningPlugin && TypelevelMimaPlugin && TypelevelCiPlugin && GitHubActionsPlugin
+
+  override def trigger = allRequirements
+
+  object autoImport {
+    lazy val tlFatalWarningsInCi = settingKey[Boolean](
+      "Convert compiler warnings into errors under CI builds (default: true)")
+  }
+
+  import autoImport._
+  import TypelevelSettingsPlugin.autoImport._
+  import GitHubActionsPlugin.autoImport._
+
+  override def globalSettings = Seq(
+    tlFatalWarningsInCi := true
```

Comment:
I have never been particularly fond of this setting varying from local development.

Reply:
True. But also, fatal warnings make development/experimentation extremely annoying.

Reply:
I generally solve that by adjusting my scalacOptions in session while in that mode.  That way it's consistent with CI by default, and I can opt out while hacking.  But I won't die on this hill if other people like it.

Reply:
Interesting. OTOH I generally solve this problem by actually paying attention to the warnings indicated in the IDE 😉

One compromise is providing a `prePR` command that emulates a CI compile.

Reply:
I like the compromise, but I thought that was the intent of the `ci` alias?

Reply:
I always thought of `ci` as what you want CI to do—run the entire test suite, run MiMa, generate docs, etc. etc. Meanwhile, `prePR` are "lighter-weight" checks you can do locally, e.g. formatting, compiling with fatal warnings, etc.

Reply:
Hmm, this is trickier to implement than I thought.
```scala
addCommandAlias(
    "prePR",
    "; root/clean; set tlFatalWarnings := tlFatalWarningsInCi.value; +root/Test/compile")
```
The problem is, how do you reset `tlFatalWarnings` to the old setting after `prePR`? Not sure if this is possible, or my sbt-fu isn't good enough anyway. Someone smarter can chime in, otherwise we can shunt this for now.

Reply:
It could be done with a custom task, but it may not be worth the trouble.